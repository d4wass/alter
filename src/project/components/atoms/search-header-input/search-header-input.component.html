<p>search-header-input works!</p>
