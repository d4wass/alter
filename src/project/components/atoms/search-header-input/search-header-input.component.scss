@use '../../../../base';

.search {
  display: flex;
  flex-direction: column;
}

.search-label {
  font-weight: base.$bold-font;
}
.search-input {
  border: none;
  outline: none;
  width: 100%;
  border-top: 10px transparent solid;

  &-noLabel {
    margin-top: 29px;
    border: none;
    outline: none;
    width: 100%;

    &::placeholder {
      color: base.$placeholder-gray;
      font-weight: base.$light-font;
    }
  }

  &::placeholder {
    color: base.$placeholder-gray;
    font-weight: base.$light-font;
  }
}
