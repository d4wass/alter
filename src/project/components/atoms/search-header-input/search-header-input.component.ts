import { Component, Input } from '@angular/core';
import { FormControl } from '@angular/forms';

@Component({
  selector: 'app-search-header-input',
  template: `
    <div class="search">
      <label for="where" class="search-label" *ngIf="labelValue">{{ labelValue }}</label>
      <input
        type="{{ typeValue }}"
        id="where"
        placeholder="{{ placeholderValue }}"
        [ngClass]="{ 'search-input': labelValue, 'search-input-noLabel': !labelValue }"
        [formControl]="control"
      />
    </div>
  `,
  styleUrls: ['./search-header-input.component.scss']
})
export class SearchHeaderInputComponent {
  @Input() labelValue!: string;
  @Input() placeholderValue!: string;
  @Input() typeValue: string = 'text';
  @Input() control!: FormControl;
}
