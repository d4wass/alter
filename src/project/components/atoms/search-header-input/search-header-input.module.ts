import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SearchHeaderInputComponent } from './search-header-input.component';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';

@NgModule({
  declarations: [SearchHeaderInputComponent],
  imports: [CommonModule, ReactiveFormsModule, FormsModule],
  exports: [SearchHeaderInputComponent]
})
export class SearchHeaderInputModule {}
