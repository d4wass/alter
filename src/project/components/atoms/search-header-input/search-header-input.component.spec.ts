import { ComponentFixture, TestBed } from '@angular/core/testing';

import { SearchHeaderInputComponent } from './search-header-input.component';

describe('SearchHeaderInputComponent', () => {
  let component: SearchHeaderInputComponent;
  let fixture: ComponentFixture<SearchHeaderInputComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [SearchHeaderInputComponent]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(SearchHeaderInputComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
