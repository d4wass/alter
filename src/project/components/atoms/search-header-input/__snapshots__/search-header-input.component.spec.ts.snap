// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeaderInputComponent should match to snapshot 1`] = `
<app-search-header-input
  control={[Function FormControl]}
  typeValue={[Function String]}
>
  <form
    class="search"
    novalidate=""
  >
    <input
      class="search-input-noLabel ng-untouched ng-pristine ng-valid"
      data-test="search-input"
      id="where"
      placeholder=""
      type="text"
    />
  </form>
</app-search-header-input>
`;
